src/Docker.DotNet/Microsoft.Net.Http.Client/BufferedReadStream.cs

```diff
@@ -159,7 +164,25 @@ private async Task EnsureBufferedAsync(CancellationToken cancel)
             if (_bufferCount == 0)
             {
                 _bufferOffset = 0;
+#if !NET45
+                bool validBuffer = Interlocked.Increment(ref _bufferRefCount) > 1;
+                try
+                {
+                    if (validBuffer)
+                    {
+                        _bufferCount = await _inner.ReadAsync(_buffer, _bufferOffset, _buffer.Length, cancel).ConfigureAwait(false);
+                    }
+                }
+                finally
+                {
+                    if ((Interlocked.Decrement(ref _bufferRefCount) == 0) && validBuffer)
```

Comment:
Shouldn't this be `if (validBuffer && ...)` so that if the caller didn't actually increment we don' try the decrement.

Reply:
No, since we've already incremented it:
`                bool validBuffer = Interlocked.Increment(ref _bufferRefCount) > 1;`

Reply:
Oh your right I totally misread that.